### PR TITLE
Add Action to store the timeout reason in the build

### PR DESCRIPTION
**Pull request by v1v**
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### What

Add `InvisibleAction` with the timeout reason operation.

### Why

Actions are persistence therefore this will help to know the timeout reason if querying the `api` or accessing programmatically the build.

### Actions

- Add tests
- Revisit why all the files were changes

### CODE REVIEW

File: src/main/java/hudson/plugins/build_timeout/operations/WriteDescriptionOperation.java [hunk 1]
```diff
@@ -68,6 +68,8 @@ public boolean perform(AbstractBuild<?, ?> build, BuildListener listener, long e
             e.printStackTrace(listener.getLogger());
         }
 
+        addAction("WriteDescriptionOperation");
```
Comment (v1v) [inline src/main/java/hudson/plugins/build_timeout/operations/WriteDescriptionOperation.java:71]:
```suggestion
        addAction(build, "WriteDescriptionOperation");

```

**Comment by krisstern** [comment]
@v1v Thanks for the PR! I will review this PR soon and get back to you ASAP.

**Comment by krisstern** [comment]
@v1v In light of recent PR's merged, would it be possible for you to review and resolve the conflicts in the following files?:
* src/test/java/hudson/plugins/build_timeout/BuildTimeoutWrapperIntegrationTest.java
* src/test/java/hudson/plugins/build_timeout/operations/AbortAndRestartOperationTest.java

**Comment by krisstern** [comment]
Code in pull request is too buggy even after initial attempts to fix. Mostly NullPointerExceptions that cannot be handled simply.

**Pull request closed without merging**